lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -97,28 +97,32 @@ pub trait InvertedIndex {
     /// If it is an unseen token, it is added to the vocabulary and a new token id is generated.
     ///
     /// The order of the tokens is preserved.
-    fn token_ids<'a>(
+    fn collect_token_ids<'a>(
         &mut self,
         str_tokens: impl IntoIterator<Item = &'a String> + 'a,
     ) -> Vec<TokenId> {
-        let vocab = self.get_vocab_mut();
         let mut token_ids = vec![];
         for token in str_tokens {
-            // check if in vocab
-            let vocab_idx = match vocab.get(token) {
-                Some(&idx) => idx,
-                None => {
-                    let next_token_id = vocab.len() as TokenId;
-                    vocab.insert(token.to_string(), next_token_id);
-                    next_token_id
-                }
-            };
-            token_ids.push(vocab_idx);
+            token_ids.push(self.token_id(token));
         }
 
         token_ids
     }
 
+    /// Translate the string tokens into token ids.
+    /// If it is an unseen token, it is added to the vocabulary and a new token id is generated.
+    fn token_id(&mut self, token_str: &str) -> TokenId {
```

Comment:
nit: maybe rename this function into something which would reflect the mutable nature. Like `register_token`.